src/sagas/pushNotification.js

```diff
@@ -0,0 +1,432 @@
+    const network = new Network(NETWORK);
+
+    const pin = yield call(showPinScreenForResult, dispatch);
+    walletService = new HathorWalletServiceWallet({
+      requestPassword: pin,
+      seed: walletUtil.getWalletWords(pin),
+      network,
+      enableWs: false,
+    });
+
+    try {
+      yield call(walletService.start.bind(walletService), {
+        pinCode: pin,
+        password: pin,
+      });
+    } catch (error) {
+      yield put(pushLoadWalletFailed({ error }));
+    }
+  } else {
+    walletService = yield select((state) => state.wallet);
+  }
+
+  yield put(pushLoadWalletSuccess({ walletService }));
+}
+
+/**
+ * This function is the actual opt-in of a user to the Push Notifications feature.
+ * It should be called when the push notifications are not loaded and/or registered.
+ * This should load the wallet on the wallet-service and register it with the deviceId.
+ * @param {{ payload: { deviceId: string } }} action
+ */
+export function* firstTimeRegistration({ payload: { deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+    return;
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.registerDevice, walletService, {
+      pushProvider: pushLib.PushNotificationProvider.ANDROID,
+      deviceId,
+      enablePush: true,
+    });
+
+    if (success) {
+      const enabledAt = Date.now();
+      STORE.setItem(pushNotificationKey.enabledAt, enabledAt);
+      STORE.setItem(pushNotificationKey.hasBeenEnabled, true);
+      yield put(pushRegisterSuccess({ enabled: true, hasBeenEnabled: true, enabledAt }));
+    } else {
+      // NOTE: theoretically, this should never happen because when the client call fails, it throws an error
+      yield put(pushRegisterFailed());
+    }
+  } catch (error) {
+    console.log('Error registering device for the first time: ', error.cause);
+    yield put(pushRegisterFailed());
+  }
+}
+
+/**
+ * This function is responsible for registering the device on the wallet-service in the event
+ * of renewing the registration.
+ */
+export function* registration({ payload: { enabled, showAmountEnabled, deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+    return;
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.registerDevice, walletService, {
+      pushProvider: pushLib.PushNotificationProvider.ANDROID,
```

Comment:
Why is this hardcoded with `ANDROID`?

Reply:
Because it is only available to android, right? But before allowing the settings for push notifications we need to verify the platform. I believe it should be the approach.